instrumentation/action_pack/lib/opentelemetry/instrumentation/action_pack/patches/action_controller/metal.rb

```diff
@@ -14,7 +14,12 @@ module Metal
             def dispatch(name, request, response)
               rack_span = OpenTelemetry::Instrumentation::Rack.current_span
               if rack_span.recording?
-                rack_span.name = "#{self.class.name}##{name}" unless request.env['action_dispatch.exception']
+                unless request.env['action_dispatch.exception']
+                  rack_span.name = case instrumentation_config[:span_naming]
+                                   when :route then "#{request.method} #{rails_route(request)}"
```

Comment:
If I am following along here, this will result in the route being computed multiple times when both `span_naming: :route` and `enable_recognize_route: true` are set. 

@robertlaurin Is there any reason why `enable_recognize_route` should _not_ rename the span instead of adding a separate `http.route` attribute?

Also, do any of the instrumentations add Rails specific code attributes? https://opentelemetry.io/docs/reference/specification/trace/semantic_conventions/span-general/#source-code-attributes

| semconv attribute | |
| --- | --- |
| `code.namespace` | `self.class.name` |
| `code.function` | `name` |

Reply:
Yes, the route is computed twice. We could compute it once and set it in a variable. But we would always be computing it, even when we don't need it.

I think having both the span name and a separate `http.route` attribute has its usefulness, especially if the span name is method + route.

As for the code attributes, it does seem adding them with controller and action would be nice. I'm happy to do that, either in this PR or in another one.

Reply:
> I think having both the span name and a separate http.route attribute has its usefulness, especially if the span name is method + route.

The latter is a question I want to answer. Should we _always_ use the recognized route as the span name instead of using the class and method name?

Reply:
That sounds good to me. I didn't want to introduce breaking change, but I agree it's a better default.
I've made the change, with an option to go back to the old format.

Reply:
@dmathieu Thank you again for your submission. I would like to give the original author(s), e.g. @robertlaurin, the opportunity to get a look at this PR and provide feedback, however he is not available for a few more days AFAIK.  

Thank you for your patience.

Reply:
SemConv backs up using the templated route as the span name:

> Given an inbound request for a route (e.g. "/users/:userID?") the name attribute of the span SHOULD be set to this route. If the route does not include the application root, it SHOULD be prepended to the span name.

https://opentelemetry.io/docs/reference/specification/trace/semantic_conventions/http/#http-server-semantic-conventions